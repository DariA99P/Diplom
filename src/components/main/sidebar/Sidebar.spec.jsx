import React from 'react';
import { shallow } from 'enzyme';

import Sidebar from '.';

describe('<Sidebar />', () => {
  let toggle;
  let wrapper;

  beforeEach(() => {
    toggle = jest.fn();
    wrapper = shallow(<Sidebar toggle={toggle} />);
  });

  describe('When rendering', () => {
    it('should renders properly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When toggle icon is clicked', () => {
    it('should toggle Sidebar collpased state', () => {
      wrapper.find('Icon').simulate('click');
      expect(toggle).toHaveBeenCalled();
    });
  });
});
