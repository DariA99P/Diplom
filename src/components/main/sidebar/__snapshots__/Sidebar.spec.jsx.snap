// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> When rendering should renders properly 1`] = `
<Sider
  className="sidebar"
  collapsed={false}
  collapsedWidth={0}
  collapsible={true}
  defaultCollapsed={false}
  prefixCls="ant-layout-sider"
  reverseArrow={false}
  style={Object {}}
  trigger={null}
  width={255}
>
  <Row
    gutter={0}
    style={
      Object {
        "backgroundColor": "var(--navy-two, #002140)",
      }
    }
  >
    <Col
      span={18}
    >
      <div
        className="logo-container"
        style={
          Object {
            "height": "100%",
            "textAlign": "center",
          }
        }
      >
        <Link
          replace={false}
          to="/"
        >
          <img
            alt="Logo"
            className="App-logo"
            src="vj_logo.png"
            style={
              Object {
                "paddingTop": "8%",
                "width": "70%",
              }
            }
          />
        </Link>
      </div>
    </Col>
    <Col
      span={6}
    >
      <Icon
        className="triggerFold"
        onClick={[Function]}
        type="menu-fold"
      />
    </Col>
  </Row>
  <Row
    gutter={0}
    justify="space-around"
    style={
      Object {
        "padding": "16px 0",
      }
    }
    type="flex"
  >
    <Search
      className="antSearch"
      enterButton={false}
      inputPrefixCls="ant-input"
      onSearch={[Function]}
      prefixCls="ant-input-search"
      style={
        Object {
          "width": 200,
        }
      }
    />
  </Row>
  <Menu
    className=""
    focusable={false}
    inlineCollapsed={false}
    mode="inline"
    prefixCls="ant-menu"
    theme="dark"
  >
    <MenuItem
      key="1"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <BulletinBoardIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Reports
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="2"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <GestureDoubleTapIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          On Demand
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="3"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <AirportIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Operations
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="4"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <AccountCardDetailsIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Program
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="5"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <WalletMembershipIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Membership
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="6"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <FinanceIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Finance
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="7"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <AccountStarIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Customer Service
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="8"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <InboxMultipleIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Service Providers
        </span>
      </div>
    </MenuItem>
    <MenuItem
      key="9"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
          }
        }
      >
        <NutritionIcon
          size="18"
        />
        <span
          style={
            Object {
              "marginLeft": "6px",
            }
          }
        >
          Catering
        </span>
      </div>
    </MenuItem>
    <SubMenu
      key="sub1"
      title={
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <SettingsIcon
            className="mdi-spin"
            size="18"
          />
          <span
            className="mdi mdi-access-point"
          />
          <span
            style={
              Object {
                "marginLeft": "6px",
              }
            }
          >
            Administration
          </span>
        </div>
      }
    >
      <MenuItem
        key="10"
      >
        Aircrafts
      </MenuItem>
      <SubMenu
        key="sub2"
        title="Airports"
      >
        <MenuItem
          key="11"
        >
          Airport Center
        </MenuItem>
      </SubMenu>
      <MenuItem
        key="12"
      >
        Companies
      </MenuItem>
      <MenuItem
        key="13"
      >
        People
      </MenuItem>
      <MenuItem
        key="14"
      >
        Black List
      </MenuItem>
      <MenuItem
        key="15"
      >
        Service Areas
      </MenuItem>
    </SubMenu>
  </Menu>
</Sider>
`;
